lib/sitemap_generator/my_zivi_google_storage_adapter.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# :nocov:
+require 'google/cloud/storage'
+
+module SitemapGenerator
+  class MyZiviGoogleStorageAdapter
+    def initialize(opts = {})
+      ENV['GOOGLE_CLOUD_CREDENTIALS'] ||= Rails.application.credentials[:gcs_credentials].as_json
+
+      opts = {
+        project_id: ENV['GCS_PROJECT_NAME'],
+        bucket: ENV['GCS_BUCKET_NAME']
+      }.merge(opts)
+
+      @bucket = opts.delete(:bucket)
```

Comment:
Why add to hash and then delete?

Reply:
Because the client is not bound to a bucket so I extract from options